source/runner/DangerfileRunner.ts

```diff
@@ -92,29 +92,50 @@ export async function runDangerfileEnvironment(
   //   return originalRequire.apply(this, arguments)
   // }
 
-  vm.run(content, filename)
-
-  const results = environment.sandbox!.results!
-  await Promise.all(
-    results.scheduled.map((fnOrPromise: any) => {
-      if (fnOrPromise instanceof Promise) {
-        return fnOrPromise
-      }
-      if (fnOrPromise.length === 1) {
-        // callback-based function
-        return new Promise(res => fnOrPromise(res))
-      }
-      return fnOrPromise()
-    })
-  )
-  return {
-    fails: results.fails,
-    warnings: results.warnings,
-    messages: results.messages,
-    markdowns: results.markdowns,
+  try {
+    vm.run(content, filename)
+
+    const results = environment.sandbox!.results!
+    await Promise.all(
+      results.scheduled.map((fnOrPromise: any) => {
+        if (fnOrPromise instanceof Promise) {
+          return fnOrPromise
+        }
+        if (fnOrPromise.length === 1) {
+          // callback-based function
+          return new Promise(res => fnOrPromise(res))
+        }
+        return fnOrPromise()
+      })
+    )
+    return {
+      fails: results.fails,
+      warnings: results.warnings,
+      messages: results.messages,
+      markdowns: results.markdowns,
+    }
+  } catch (error) {
+    console.error("Unable to evaluate the Dangerfile")
+    return resultsForCaughtError(filename, content, error)
   }
 }
 
+/** Returns Markdown results to post if an exception is raised during the danger run */
+const resultsForCaughtError = (file: string, contents: string, error: Error): DangerResults => {
+  const failure = `Danger failed to run \`${file}\`.`
+  const errorMD = `## Error ${error.name}
+\`\`\`
+${error.message}
+${error.stack}
+\`\`\`
+### Dangerfile
+\`\`\`
```

Comment:
I removed the ts for now - will look into showing which line caused the problem next. 👍